pepper-apis/src/test/java/org/broadinstitute/ddp/service/Auth0LogEventServiceTest.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.ddp.service;
+
+import static org.broadinstitute.ddp.constants.Auth0LogEventTestConstants.AUTH0_LOG_EVENT_TESTDATA1;
+import static org.broadinstitute.ddp.constants.Auth0LogEventTestConstants.AUTH0_LOG_EVENT_TESTDATA2;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.CLIENT_ID;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.EMAIL;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.IP;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.TYPE;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.USER_AGENT;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.USER_ID;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.USER_NAME;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.resolveStringValue;
+import static org.broadinstitute.ddp.util.TestUtil.readJSONFromFile;
+import static org.junit.Assert.assertEquals;
+
+import java.io.FileNotFoundException;
+
+
+import org.junit.Test;
+
+
+public class Auth0LogEventServiceTest {
+
+    final Auth0LogEventService auth0LogEventService = new Auth0LogEventService();
+
+    @Test
+    public void testAuth0LogEventsParse() throws FileNotFoundException {
+        final var logEvents = auth0LogEventService.parseAuth0LogEvents(readJSONFromFile(AUTH0_LOG_EVENT_TESTDATA1).toString());
+
+        assertEquals(2, logEvents.size());
+
+        final var logEvent = logEvents.get(0);
+
+        assertEquals("s", logEvent.get(TYPE.nodeName()).getAsString());
+        assertEquals("auth0|XXX_user_id_XXX", logEvent.get(USER_ID.nodeName()).getAsString());
```

Comment:
Will be interesting to see what `user_id` is for `logEvents.get(1)`.

Reply:
I added the check of logEvents.get(1) and found some problem in GsonRecursiveReader. Fixed this now.

Reply:
The problem was that the reader when going through elements of a node process each node recursively immediately as it encountered. I fixed it by splitting into 2 for-circles: 1st one reads values of each element (without recursive processing), 2nd - process each element recursively, if possible. Now it reads the nodes as expected.
Thanks a lot for suggestion to check this!